app/models/game.rb

```diff
@@ -4,40 +4,68 @@ class Game < ApplicationRecord
-after_create :populate_board!
+  after_create :populate_board!
 
- def populate_board!
-   # black pieces
-   (0..7).each do |i|
-     Pawn.create(game_id: id, x_space: i, y_space: 1, color: "black")
-   end
-   Rook.create(game_id: id, x_space: 0, y_space: 0, color: "black")
-   Rook.create(game_id: id, x_space: 7, y_space: 0, color: "black")
+   def populate_board!
+     # black pieces
+     (0..7).each do |i|
+       Pawn.create(game_id: id, x_space: i, y_space: 1, color: "black")
+     end
+     Rook.create(game_id: id, x_space: 0, y_space: 0, color: "black")
+     Rook.create(game_id: id, x_space: 7, y_space: 0, color: "black")
 
-   Knight.create(game_id: id, x_space: 1, y_space: 0, color: "black")
-   Knight.create(game_id: id, x_space: 6, y_space: 0, color: "black")
+     Knight.create(game_id: id, x_space: 1, y_space: 0, color: "black")
+     Knight.create(game_id: id, x_space: 6, y_space: 0, color: "black")
 
-   Bishop.create(game_id: id, x_space: 2, y_space: 0, color: "black")
-   Bishop.create(game_id: id, x_space: 5, y_space: 0, color: "black")
+     Bishop.create(game_id: id, x_space: 2, y_space: 0, color: "black")
+     Bishop.create(game_id: id, x_space: 5, y_space: 0, color: "black")
 
-   Queen.create(game_id: id, x_space: 3, y_space: 0, color: "black")
-   King.create(game_id: id, x_space: 4, y_space: 0, color: "black")
+     Queen.create(game_id: id, x_space: 3, y_space: 0, color: "black")
+     King.create(game_id: id, x_space: 4, y_space: 0, color: "black")
 
-    # white pieces
-     (0..7).each do |i|
-       Pawn.create(game_id: id, x_space: i, y_space: 6, color: "white")
-     end
-    Rook.create(game_id: id, x_space: 0, y_space: 7, color: "white")
-   Rook.create(game_id: id, x_space: 7, y_space: 7, color: "white")
+      # white pieces
+       (0..7).each do |i|
+         Pawn.create(game_id: id, x_space: i, y_space: 6, color: "white")
+       end
+      Rook.create(game_id: id, x_space: 0, y_space: 7, color: "white")
+     Rook.create(game_id: id, x_space: 7, y_space: 7, color: "white")
+
+     Knight.create(game_id: id, x_space: 1, y_space: 7, color: "white")
+     Knight.create(game_id: id, x_space: 6, y_space: 7, color: "white")
+
+     Bishop.create(game_id: id, x_space: 2, y_space: 7, color: "white")
+     Bishop.create(game_id: id, x_space: 5, y_space: 7, color: "white")
+
+     Queen.create(game_id: id, x_space: 3, y_space: 7, color: "white")
+     King.create(game_id: id, x_space: 4, y_space: 7, color: "white")
+   end
 
-   Knight.create(game_id: id, x_space: 1, y_space: 7, color: "white")
-   Knight.create(game_id: id, x_space: 6, y_space: 7, color: "white")
+  def check?(color)
+    black_king = game.pieces.find_by(color: "black", type: "King")
+    white_king = game.pieces.find_by(color: "white", type: "King")
+    
+    black_piece = game.pieces.find_by(color: "black", type: "Piece")
```

Comment:
I see that you're trying to fetch all the pieces, however Piece is not a type. Only King, Queen, Rook, Bishop, etc. You should replace `find_by` with `where` (it returns an array of pieces where as `find_by` only returns a single object.) When you put `where`, you are able to remove the type and just have the color passed in.

Reply:
Thank you so much. I knew I should have looked at it again this morning before submitting. 🤣 Will fix and submit tomorrow.